client/src/store/cards/cards.reducer.ts

```diff
@@ -45,15 +46,40 @@ export const cardsReducer = (
         enemyCardCount: state.enemyCardCount + action.payload.amount
       };
 
+    case CardsActionTypes.MoveMyCardsWithinArray:
+      state.myCards.splice(action.payload.currentIndex, 0, state.myCards.splice(action.payload.previousIndex, 1)[0]);
```

Comment:
prefer to generate new array (may be with spread)

Reply:
there is no need to return a new array